Dockerfile

```diff
@@ -98,16 +98,28 @@ ENV CDM_SHARED_DIR=/cdm_shared_workspace
 RUN mkdir -p ${CDM_SHARED_DIR} && chmod -R 777 ${CDM_SHARED_DIR}
 RUN chown -R spark_user:spark $CDM_SHARED_DIR
 
+# Set a directory for hosting Hive metastore files - defined in config/hive-site-template.xml
+ENV HIVE_METASTORE_DIR=$CDM_SHARED_DIR/hive_metastore
+RUN mkdir -p ${HIVE_METASTORE_DIR}
+RUN chown -R spark_user:spark $HIVE_METASTORE_DIR
+
 # Set a directory for hosting Jupyterhub db and cookie secret
 ENV JUPYTERHUB_SECRETS_DIR=/jupyterhub_secrets
 RUN mkdir -p ${JUPYTERHUB_SECRETS_DIR}
 RUN chown -R spark_user:spark ${JUPYTERHUB_SECRETS_DIR}
 
+# Set a directory for hosting postgres db data - used by Spark to store Delta tables
+ENV JUPYTERHUB_POSTGRES_DIR=/cdm-postgres
+RUN mkdir -p ${JUPYTERHUB_POSTGRES_DIR}
+RUN chown -R spark_user:spark ${JUPYTERHUB_POSTGRES_DIR}
+
 # Allow spark_user to use sudo without a password
 # TODO: use `sudospawner` in JupyterHub to avoid this (https://jupyterhub.readthedocs.io/en/stable/howto/configuration/config-sudo.html)
 RUN echo "spark_user ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
 
 # Switch back to non-root user
-USER spark_user
+# Facing permission errors when accessing the Docker API as a non-root user
+#USER spark_user
+
```

Comment:
So is this dockerfile for the spawner, and hence running at root?

Reply:
hmmm I currently use the same dockerfile (image) for both jupyterhub server and the spawner (user). The hub needs to run docker socket so that it can spin up containers for users. That's where I encounter permission issues with spark_user. Boris actually suggests that we use a simpler image for the server. And use this docker file specifically for the user. I think that makes sense to do. But maybe later work.

Reply:
Yeah, I would def not run the user containers as root, that makes sense. Later work is fine

Reply:
👍 just want to get something that works for now. Will revisit.